client/server.go

```diff
@@ -28,6 +28,19 @@ const (
 	ServerStateClosed     = int32(iota)
 )
 
+// The RequestHandler invocation context. Each invocation of a given RequestHandler will be passed
+// one instance of a RequestHandlerContext, that remains the same between invocations. This allows
+// handlers to become stateful if required.
+type RequestHandlerContext map[string]interface{}
```

Comment:
I don't know if this helps make the code more readable or if it does the opposite. When I saw this type being used in the functions below I had to search for it while if the `map` type was referenced directly I wouldn't have to search for the type definition. 

The comment is very helpful though so maybe we could get rid of the alias and move this comment to the field definition of the server connection struct?

Reply:
Indeed it is a bit less readable for us developers of this library... but what about users that are going to implement custom `RequestHandler`s?

By using `RequestHandlerContext`, the signature of a typical `RequestHandler` function is:

```
func MyCustomHandler(request *frame.Frame, conn *CqlServerConnection, ctx RequestHandlerContext) (response *frame.Frame)
```

With that signature, it's easy to look up the documentation of `RequestHandlerContext` and understand how to use it. However if we use the below signature instead:

```
func MyCustomHandler(request *frame.Frame, conn *CqlServerConnection, ctx map[string]interface{}) (response *frame.Frame)
```

Then the user wouldn't know where to look for documentation regarding `ctx` and how to use it. That's why I am a bit reluctant to implement your suggestion.

Reply:
Hmm good point but what about moving the comment to that method/function instead of the struct then

Reply:
Maybe even better than this, what about actually creating a type for this context and expose a method to add and read key values from the context? Might be overengineering though

Reply:
I like it, will do as you suggested.